flink-clients/src/main/java/org/apache/flink/client/cli/ArtifactFetchOptionsInternal.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.client.cli;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.configuration.ConfigOption;
+import org.apache.flink.configuration.ConfigOptions;
+
+import java.util.List;
+
+/** Artifact Fetch options. */
+@Internal
+public class ArtifactFetchOptionsInternal {
+
+    public static final ConfigOption<List<String>> COMPLETE_LIST =
+            ConfigOptions.key("$internal.user.artifacts.complete-list")
```

Comment:
This is currently unused. The current implementation modifies the passed `user.artifacts.artifact-list` config option, which was more straightforward regarding the impl but maybe confusing/magical for users, so it can be changed of the current behavior would be a no go. Otherwise, this can be removed.

Personally I do not think modifying the given config option would be risky, IMO it can be justified to show the final deployed state in the config.

Reply:
That is a good point, follow the same practice makes perfect sense!